docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,141 @@
+
+The *interblockchain communication protocol* takes an orthogonal approach to a differently formulated version of the problem: enabling safe, reliable interoperation of a network of heterogeneous distributed ledgers, arranged in an unknown topology, which can diversify, develop, and rearrange independently of each other or of a particular imposed topology or state machine design. In a wide, dynamic network of interoperating chains, sporadic Byzantine faults are expected, so the protocol must also detect, mitigate, and contain the potential damage of Byzantine faults in accordance with the requirements of the applications & blockchains involved. For a longer list of design principles, see [here](./2_IBC_DESIGN_PRINCIPLES.md).
+
+To faciliate this heterogeneous interoperation, the interblockchain communication protocol takes a "bottom-up" approach, specifying the set of requirements, functions, and properties necessary to implement interoperation between two ledgers, and then specifying different ways in which multiple interoperating ledgers might be composed which preserve the requirements of higher-level protocols and occupy different points in the safety/speed tradeoff space. IBC thus presumes nothing about and requires nothing of the overall network topology, and of the implementing ledgers requires only that a known, minimal set of functions are available and properties fulfilled.
+
+IBC is an end-to-end, connection-oriented, stateful protocol for reliable, ordered, authenticated communication between modules on separate distributed ledgers. IBC implementations are expected to be co-resident with higher-level modules and protocols on the host ledger. Ledgers hosting IBC must provide a certain set of functions for consensus transcript verification and accumulator proof generation, and IBC packet relayers (off-chain processes) are expected to have access to network protocols and physical datalinks as required to read the state of one ledger and submit data to another.
+
+### Dataflow layers
+
+```
++--------------------------+                           +--------------------------+
+| Distributed Ledger A     |                           | Distributed Ledger B     |
+|                          |                           |                          |
+| +----------+     +-----+ |        +---------+        | +-----+     +----------+ |
+| | Module A | <-> | IBC | | <----> | Relayer | <----> | | IBC | <-> | Module B | |
+| +----------+     +-----+ |        +---------+        | +-----+     +----------+ |
++--------------------------+                           +--------------------------+
+```
+
+## Scope
+
+IBC handles authentication, transport, and ordering of structured data packets relayed between modules on separate ledgers. The protocol is intended to be in simultaneous use between any number of modules on any number of ledgers over arbitrarily structured underlying networks.
```

Comment:
The second sentence is not very clear. Does it make sense to initially constrain IBC on inter-module communication between two ledgers?However, we can have several modules on ledger A communicating over IBC with modules on ledger B.

Reply:
Right, that's confusing. I'm trying to emphasize that it's safe to run multiple instances of the IBC protocol at the same time between many different ledgers which are all connected to each other.

Tried to clarify.